crates/rpc/rpc/src/eth/api/mod.rs

```diff
@@ -448,7 +467,258 @@ struct EthApiInner<Provider, Pool, Network> {
     pending_block: Mutex<Option<PendingBlock>>,
     /// A pool dedicated to blocking tasks.
     blocking_task_pool: BlockingTaskPool,
+    /// Cache for block fees history
+    fee_history_cache: FeeHistoryCache,
     /// An http client for communicating with sequencers.
     #[cfg(feature = "optimism")]
     http_client: reqwest::Client,
 }
+
+/// Settings for the [FeeHistoryCache](crate::eth::FeeHistoryCache).
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct FeeHistoryCacheConfig {
```

Comment:
I'd like to move all of the history adjacent types to a new file, fee_history

Reply:
Ok! Moved them to separate file! hope it works